xen-bugtool

```diff
@@ -1396,9 +1406,9 @@ def dump_xapi_rrds(cap):
             rrd = urlopen("http://localhost/vm_rrd?session_id=%s&uuid=%s" % (session._session, vm["uuid"]))
             try:
                 i, _, _ = select([rrd], [], [], 5.0)
-                if len(i) == 1:
-                    data['xapi_rrd-%s' % vm['uuid']] = {'cap': cap,
-                                                        'output': StringIOmtime(rrd.read())}
+                if len(i) == 1:  # pragma: no cover # TODO: Write a unit test
+                    name = "xapi_rrd-%s.out" % vm['uuid']
+                    archive.add_path_with_data(name, StringIOmtime(rrd.read()))
```

Comment:
Hmm, I'm not sure about this. It's bound to go wrong when someone doesn't realize that `archive` is a global variable and is being used elsewhere. The name `archive` is also used elsewhere as a local variable. This also doesn't respect the capability sizes.

I think it might be better to convert `dump_xapi_rrds()` into a function that calls `func_output()` once for each VM record where the function to call is a lambda function that simply returns the output retrieved here.
E.g. something like:

```
rrddata = rrd.read()
func_output(CAP_PERSISTENT_STATS, lambda _: rrddata)
```

and then call `dump_xapi_rrds()` directly rather than via `func_output`.